### PR TITLE
BUGFIX: Avoid migration failure if files are not existing in FS

### PR DESCRIPTION
**Pull request by kaystrobach**
**What I did**

This fix migrates non existing files into a separate table, so that they can be fixed lateron.
Having this change creates a new table, named
typo3_flow_resource_resource_backupfiles, which contains file entries, which do not comply
with the contraint rules.

**How I did it**

Adjust migration

**How to verify it**

* add a file into the resources table, remove the actual file from the FS, then run migrations
* run migrations esp. *20141015125841*

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)

### CODE REVIEW

**Comment by kdambekalns** [comment]
@kaystrobach Can't you fix this by running `resource:clean` before doing the migration?

**Comment by kaystrobach** [comment]
hey @kdambekalns this has sadly not worked out.

File: TYPO3.Flow/Migrations/Mysql/Version20141015125841.php [hunk 1]
```diff
@@ -67,6 +70,8 @@ public function postUp(Schema $schema)
                 continue;
             }
 
+            $this->write(print_r(array('persistent', $mediaType, $md5, $filesize, $resourceInfo['persistence_object_identifier']), true));
```
Comment (kdambekalns) [inline TYPO3.Flow/Migrations/Mysql/Version20141015125841.php]: This looks like debug output, is that intentionally left here?

Comment (kaystrobach) [reply TYPO3.Flow/Migrations/Mysql/Version20141015125841.php]: Can be removed, i used it during the migration.

Comment (kaystrobach) [reply TYPO3.Flow/Migrations/Mysql/Version20141015125841.php]: Not creating the table is also possible. One solution would be to check if the migration can be applied before it starts and do not fail at 95%

**Comment by kaystrobach** [comment]
Then we could also add the hint how to remove data, which is not valid anymore ...

**Pull request closed without merging**